dashboard/app/views/levels/_dialog.html.haml

```diff
@@ -68,131 +68,14 @@
-
-    // Clicking the okay button in the dialog box dismisses it.
-    $(".modal-body #ok-button").click(function () {
-      dialog.hide();
-    });
-
-    dialog.show();
-  }
-
-  function adjustScroll() {
-    if (adjustedScroll) {
-      return;
-    }
-
-    var win = $(window);
-    var el = $('.mainblock');
-    var winPos = win.scrollTop() + win.height();
-    var elPos = el.offset().top + el.height() - 10;
-
-    if (winPos < elPos) {
-      $('html, body').animate({
-        scrollTop: $(".submitButton:first").offset().top - 10
-      }, 1000);
-    }
-
-    adjustedScroll = true;
-  }
-
-  // TODO(dave): Dashboard shouldn't be reaching into the internal implementation of
-  // individual levels. Instead levels should call appOptions.onAttempt.
-  $('.submitButton').click(function () {
-    var submitButton = $('.submitButton');
-    if (submitButton.attr('disabled')) {
-      return;
-    }
-
-    // Avoid multiple simultaneous submissions.
-    submitButton.attr('disabled', true);
-
-    var onComplete = function(willRedirect) {
-      if (!willRedirect) {
-        $('.submitButton').attr('disabled', false);
-      }
-    };
-
-    processResults(onComplete);
-  });
-
-  /**
-   * Process the solution to the puzzle submitted by the user.
-   * @param {function(boolean)} onComplete Optional callback function to call when
-   *     the server call completes, which receives a boolean indicating whether
-   *     the browser will redirect to a new location after it is called.
-   */
-  // TODO(dave): move this logic into appOptions.onAttempt for levels of type
-  // external (including pixelation), multi, match, and any others
-  // which render 'levels/dialog'.
-  function processResults(onComplete) {
-    var results = getResult();
-    var response = results['response'];
-    var result = results['result'];
-    var errorType = results['errorType'];
-
-    if (!result) {
-      showDialog(errorType || "error");
-      if (!skipSound) {
-        CDOSounds.play('failure');
-      }
-    } else {
-      if (!skipSound) {
-        CDOSounds.play('success');
-      }
-    }
-
-    sendReport({
-      program: response,
-      fallbackResponse: '#{ @fallback_response.to_json}',
-      callback: '#{ @callback}',
-      app: '#{ app}',
-      level: '#{ @level.level_num}',
-      result: result,
-      testResult: result ? 100 : 0,
-      onComplete: function () {
-        var willRedirect = !!lastServerResponse.nextRedirect;
-        onComplete && onComplete(willRedirect);
-
-        if (lastServerResponse.videoInfo)
-        {
-          showVideoDialog(lastServerResponse.videoInfo);
-        } else if (lastServerResponse.nextRedirect) {
-          if (#{ data['options'].try(:[], 'skip_dialog').blank?}) {
-            showDialog("success");
-          } else {
-            window.location.href = lastServerResponse.nextRedirect;
-          }
-        }
-      }
-    });
-  }
+  appOptions.dialog = {
```

Comment:
Nit: let's construct this hash in Ruby then `.to_json` it like we do for the main `appOptions`.